contract/controllers/main.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, http
+from odoo.exceptions import AccessError, MissingError
+from odoo.http import request
+from odoo.addons.portal.controllers.portal import CustomerPortal,\
+    pager as portal_pager
+
+
+class PortalContract(CustomerPortal):
+
+    def _prepare_portal_layout_values(self):
+        values = super()._prepare_portal_layout_values()
+        model = 'contract.contract'
+        if request.env[model].check_access_rights('read', raise_exception=False):
+            values['contract_count'] = request.env[model].search_count([])
+        else:
+            values['contract_count'] = 0
+        return values
+
+    def _contract_get_page_view_values(self, contract, access_token, **kwargs):
+        values = {
+            'page_name': 'Contracts',
+            'contract': contract,
+        }
+        return self._get_page_view_values(
+            contract, access_token, values, 'my_contracts_history', False, **kwargs)
+
+    def _get_filter_domain(self, kw):
+        return []
+
+    @http.route(['/my/contracts', '/my/contracts/page/<int:page>'],
+                type='http', auth="user", website=True)
+    def portal_my_contracts(self, page=1, date_begin=None, date_end=None,
+                            sortby=None, **kw):
+        values = self._prepare_portal_layout_values()
+        contract_obj = request.env['contract.contract']
+        domain = self._get_filter_domain(kw)
+        searchbar_sortings = {
+            'date': {'label': _('Date'), 'order': 'recurring_next_date desc'},
+            'name': {'label': _('Name'), 'order': 'name desc'},
+            'code': {'label': _('Reference'), 'order': 'code desc'},
+        }
+        # default sort by order
+        if not sortby:
+            sortby = 'date'
+        order = searchbar_sortings[sortby]['order']
+        archive_groups = self._get_archive_groups('contract.contract', domain)
+        if date_begin and date_end:
+            domain += [
```

Comment:
Can you test this?

Reply:
@victoralmau have double-checked that all this is dead code that comes from old times. In v14 it disappeared after https://github.com/odoo/odoo/commit/e75086000ee4317b2ad2994db5d906fbcbd5081f. All this can be safely removed, which will be easier than testing it.